tokio/src/runtime/task/waker.rs

```diff
@@ -44,12 +44,38 @@ impl<S> ops::Deref for WakerRef<'_, S> {
     }
 }
 
+cfg_trace! {
+    macro_rules! trace {
+        ($harness:expr, $op:expr) => {
+            if let Some(id) = $harness.id() {
```

Comment:
it looks like the intention here is to avoid emitting the event if the task is not being tracked? i wonder if we actually want to do this _inside_ of the tracing event macro, to handle cases where the event is disabled by one of tracing's filtering fast paths. e.g., if the user compiles tracing with one of the "static_max_level_xxx" feature flags, the event macro will be completely empty and won't compile to any code, but we'll still dereference the trailer and check if it has a task ID. 

this is probably an unnecessary microoptimzation, though, and we could do it later --- moving it inside the macro will be much easier when https://github.com/tokio-rs/tracing/pull/1393 is published, anyway.

Reply:
Yea, I'd really want to be to check tracing if the event would be emitted, _and_ check if the ID is `Some`, in the same `if` statement. I don't want to emit an event if we're not tracking the task either.